server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4796,6 +4815,40 @@ private boolean localDirectoryContains(Directory localDirectory, String file, lo
         return false;
     }
 
+    private void beforeSegmentDownload(RemoteRefreshSegmentTracker downloadStatsTracker, long incomingFileSize) {
+        if (!indexSettings.isRemoteStoreEnabled()) {
+            return;
+        }
+        downloadStatsTracker.incrementTotalDownloadsStarted();
+        downloadStatsTracker.addDownloadBytesStarted(incomingFileSize);
+    }
+
+    private void afterSegmentDownloadCompleted(
+        RemoteRefreshSegmentTracker downloadStatsTracker,
+        long downloadedFileSize,
+        long startTimeInNs
+    ) {
+        if (!indexSettings.isRemoteStoreEnabled()) {
+            return;
+        }
+        long currentTimeInNs = System.nanoTime();
+        downloadStatsTracker.updateLastDownloadTimestampMs(System.currentTimeMillis());
+        downloadStatsTracker.incrementTotalDownloadsSucceeded();
+        downloadStatsTracker.addDownloadBytes(downloadedFileSize);
+        downloadStatsTracker.addDownloadBytesSucceeded(downloadedFileSize);
+        long timeTakenInMS = Math.max(1, TimeValue.nsecToMSec(currentTimeInNs - startTimeInNs));
+        downloadStatsTracker.addDownloadTime(timeTakenInMS);
```

Comment:
should this time dictate individual file download time or overall replica sync from remote store time?

Reply:
Would be addressed with adding new metrics as mentioned above